docs/fsharp/style-guide/component-design-guidelines.md

```diff
@@ -0,0 +1,873 @@
+
+### Add XML documentation comments to your code
+
+XML documentation on public APIs ensure that users can get great Intellisense and Quickinfo when using these types and members, and enable building documentation files for the library. See the [XML Documentation](../language-reference/xml-documentation.md) about various xml tags that can be used for additional markup within xmldoc comments.
+
+```fsharp
+/// A class for representing (x,y) coordinates
+type Point =
+
+    /// Computes the distance between this point and another
+    member DistanceTo : otherPoint:Point -> float
+```
+
+### Consider using explicit signature files (.fsi) for stable library and component APIs
+
+Using explicit signatures files in an F# library provides a succinct summary of public API, which both helps to ensure that you know the full public surface of your library, as well as provides a clean separation between public documentation and internal implementation details. Note that signature files add friction to changing the public API, by requiring changes to be made in both the implementation and signature files. As a result, signature files should typically only be introduced when an API has become solidified and is no longer expected to change significantly.
+
+### Always follow best practices for using strings in .NET
+
+Follow [Best Practices for Using Strings in .NET](../../standard/base-types/best-practices-strings.md) guidance. In particular, always explicitly state *cultural intent* in the conversion and comparison of strings (where applicable).
+
+## Guidelines for F#-Facing Libraries
+
+In this section, we present recommendations for developing public F#-facing libraries; that is, libraries exposing public APIs that are intended to be consumed by F# developers. There are a variety of library-design recommendations applicable specifically to F#. In the absence of specific recommendations below, the .NET Library Design Guidelines are the fallback guidance.
+
+### Naming Conventions
+
+#### Use .NET naming and capitalization conventions
+
+The following table follows .NET naming and capitalization conventions. There are small additions to also include F# constructs.
+
+| Construct | Case | Part | Examples | Notes |
+|-----------|------|------|----------|-------|
+| Concrete types | PascalCase | Noun/ adjective | List, Double, Complex | Concrete types are structs, classes, enumerations, delegates, records, and unions. Though type names are traditionally lowercase in OCaml, F# has adopted the .NET naming scheme for types.
+| DLLs           | PascalCase |                 | Fabrikam.Core.dll |  |
+| Union tags     | PascalCase | Noun | Some, Add, Success | Do not use a prefix in public APIs. Optionally use a prefix when internal, such as `type Teams = TAlpha | TBeta | TDelta.` |
```

Comment:
Unfortunately, GitHub doesn't render `` `type Teams = TAlpha | TBeta | TDelta.` |`` fragment properly. Will docs.microsoft.com render it?